sycl/include/sycl/ext/oneapi/annotated_arg/annotated_arg.hpp

```diff
@@ -76,6 +76,9 @@ __SYCL_TYPE(annotated_arg) annotated_arg<T *, detail::properties_t<Props...>> {
 public:
   static_assert(is_property_list<property_list_t>::value,
                 "Property list is invalid.");
+  static_assert(
+      validate_annotated_type<annotated_arg<T *, property_list_t>>::value,
+      "The property list contains invalid property.");
```

Comment:
I hope in a future update we can make this better? Which property is invalid? Why is it invalid? Not gating on this.

Reply:
I see better errors are bellow.